test/recipes/02-test_errstr.t

```diff
@@ -0,0 +1,41 @@
+#! /usr/bin/env perl
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use OpenSSL::Test;
+use POSIX;
+
+setup('test_errstr');
+
+# We actually have space for up to 4095 error messages,
+# numerically speaking...  but we're currently only using
+# numbers 1 through 127.
+plan tests => 128;
+
+my $i;
+
+for ($i = 1; $i < 128; $i++) {
+    # We know that the system reasons are in library 2
+    my @oerr = run(app([ qw(openssl errstr), sprintf("2%06x", $i) ]),
+                   capture => 1);
+    $oerr[0] =~ s|\R$||;
+    $oerr[0] =~ s|.*system library:||g; # The actual message is last
+
+    my $perr = strerror($i);
+    if ($perr =~ m|Unknown error $i|) {
+        ok($oerr[0] eq 'unknown', "($i) '$oerr[0]' == 'unknown' ('$perr')");
+    } else {
+        ok($oerr[0] eq $perr, "($i) '$oerr[0]' == '$perr'");
```

Comment:
I'm wondering if we might see failures in this test for some locales with very verbose error messages (sufficient to go over the 4k buffer limit).

Reply:
The OpenSSL app doesn't set locale, which is effectively accepting the "C" locale. Our tests use that locale too, so I believe this test is safe in that regard.